### PR TITLE
Bug 1796514: Adds openAPIV3Schema to NetworkAttachmentDefinitions

### PR DESCRIPTION
**Pull request by dougbtv**
To fit requirements for having a structural schema to CRDs.

### CODE REVIEW

**Comment by danwinship** [comment]
/lgtm cancel

```level=error msg="Cluster operator network Degraded is True with ApplyOperatorConfig: Error while updating operator configuration: could not apply (apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition) /network-attachment-definitions.k8s.cni.cncf.io: could not create (apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition) /network-attachment-definitions.k8s.cni.cncf.io: CustomResourceDefinition.apiextensions.k8s.io \"network-attachment-definitions.k8s.cni.cncf.io\" is invalid: spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:\"v1\", Served:true, Storage:true, Schema:(*apiextensions.CustomResourceValidation)(0xc00d133d28), Subresources:(*apiextensions.CustomResourceSubresources)(nil), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version schemas may not all be set to identical values (top-level validation should be used instead)"```

**Comment by dougbtv** [comment]
Thanks for finding that one Dan, appreciate it. I've got a new commit pushed up that doesn't use the "versions:" method, and instead pushes it up to top level validation, let's see if it's happier with that one.

**Comment by dcbw** [comment]
@dougbtv 
```
level=error msg="Cluster operator network Degraded is True with ApplyOperatorConfig: Error while updating operator configuration: could not apply (apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition) /network-attachment-definitions.k8s.cni.cncf.io: could not create (apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition) /network-attachment-definitions.k8s.cni.cncf.io: CustomResourceDefinition.apiextensions.k8s.io \"network-attachment-definitions.k8s.cni.cncf.io\" is invalid: [spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion(nil): must have exactly one version marked as storage version, status.storedVersions: Invalid value: []string(nil): must have at least one stored version]"
```

**Comment by dougbtv** [comment]
Gahhh, I think I had wrong indentation there after applying to a local cluster.... let's see if this one is good.

**Comment by dougbtv** [comment]
/test e2e-gcp

Looks like a flake.

**Comment by dougbtv** [comment]
/test e2e-gcp

Still looks like a flake to me, with other operators still updating, such as:

```
level=fatal msg="failed to initialize the cluster: Some cluster operators are still updating: authentication, console, csi-snapshot-controller, image-registry, ingress, kube-storage-version-migrator, monitoring"
--



```

**Comment by dougbtv** [comment]
/test e2e-gcp

Looked like GCP error allocating resources.

File: bindata/network/additional-networks/crd/001-crd.yaml [hunk 1]
```diff
@@ -5,11 +5,23 @@ metadata:
...
   group: k8s.cni.cncf.io
-  version: v1
   scope: Namespaced
   names:
     plural: network-attachment-definitions
     singular: network-attachment-definition
     kind: NetworkAttachmentDefinition
     shortNames:
     - net-attach-def
+  versions:
```
Comment (abhat) [inline bindata/network/additional-networks/crd/001-crd.yaml]: please add `preserveUnknownFields` to this so that `oc explain` will work.

**Pull request merged**